### PR TITLE
ACS-3601: Add unlink e2es

### PR DESCRIPTION
**Pull request by KikoUK**
https://alfresco.atlassian.net/browse/ACS-3601

### CODE REVIEW

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RuleSetLinksTests.java [hunk 1]
```diff
@@ -310,4 +308,77 @@ public void linkToParentNodeWithRules()
...
+        final RestRuleSetLinkModel request = new RestRuleSetLinkModel();
+        request.setId(ruleFolder.getNodeRef());
+        final RestRuleSetLinkModel ruleLink = restClient.authenticateUser(user).withCoreAPI().usingNode(folder).createRuleLink(request);
+
+        STEP("Unlink the rule set");
+        restClient.authenticateUser(user).withCoreAPI().usingNode(folder).unlinkRuleSet(ruleSetId);
+
+        STEP("Assert unlink result");
+        restClient.assertStatusCodeIs(NO_CONTENT);
+    }
```
Comment (tpage-alfresco) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RuleSetLinksTests.java:344]: Is it worth calling the GET to check if the rule set is still linked?

**Review by tpage-alfresco — APPROVED** [review]
LGTM

**Review by gevagel — APPROVED** [review]

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RuleSetLinksTests.java [hunk 2]
```diff
@@ -310,4 +308,87 @@ public void linkToParentNodeWithRules()
...
+                .getListOfRuleSets();
+        ruleSets.assertThat().entriesListCountIs(1);
+        final String ruleSetId = ruleSets.getEntries().get(0).onModel().getId();
+
+        STEP("Link to a rule folder");
+        final RestRuleSetLinkModel request = new RestRuleSetLinkModel();
+        request.setId(ruleFolder.getNodeRef());
+        final RestRuleSetLinkModel ruleLink = restClient.authenticateUser(user).withCoreAPI().usingNode(folder).createRuleLink(request);
+
+        STEP("Unlink the rule set");
```
Comment (krdabrowski) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RuleSetLinksTests.java:339]: Before unlinking you didn't check if the link really exists (linking result code or if linkedTo is true).

Comment (krdabrowski) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/RuleSetLinksTests.java:339]: Fair enough - I agree

**Pull request merged**